tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -16,31 +16,40 @@ import (
 var _ = Describe("odo devfile push command tests", func() {
 	var namespace, context, cmpName, currentWorkingDirectory string
 	var sourcePath = "/projects/nodejs-web-app"
+	var cliRunner helper.CliRunner
 
-	// TODO: all oc commands in all devfile related test should get replaced by kubectl
-	// TODO: to goal is not to use "oc"
-	oc := helper.NewOcRunner("oc")
+	// Using program commmand according to cliRunner in devfile
+	if os.Getenv("KUBERNETES") == "true" {
+		cliRunner = helper.NewKubectlRunner("kubectl")
+	} else {
+		cliRunner = helper.NewOcRunner("oc")
+	}
 
 	// This is run after every Spec (It)
 	var _ = BeforeEach(func() {
 		SetDefaultEventuallyTimeout(10 * time.Minute)
-		namespace = helper.CreateRandProject()
 		context = helper.CreateNewContext()
-		currentWorkingDirectory = helper.Getwd()
-		cmpName = helper.RandString(6)
-
-		helper.Chdir(context)
-
 		os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
 
 		// Devfile push requires experimental mode to be set
 		helper.CmdShouldPass("odo", "preference", "set", "Experimental", "true")
+
+		if os.Getenv("KUBERNETES") == "true" {
+			helper.LocalKubeconfigSet(context)
+		}
+		namespace = cliRunner.CreateRandNamespaceProject()
+		currentWorkingDirectory = helper.Getwd()
+		cmpName = helper.RandString(6)
+		helper.Chdir(context)
 	})
 
 	// Clean up after the test
 	// This is run after every Spec (It)
 	var _ = AfterEach(func() {
-		helper.DeleteProject(namespace)
+		cliRunner.DeleteNamespaceProject(namespace)
+		if os.Getenv("KUBERNETES") == "true" {
+			os.Unsetenv("KUBECONFIG")
```

Comment:
This unset might be causing the problems that you have on OpenShift CI.
OpenShift CI sets custom KUBECONFIG, you change it in your `LocalKubeconfigSet` but then instead of restoring its original value, you just unset it.

Reply:
Yes, This was the main reason it seems. As per our discussion I have updated the change related to `KUBECONFIG`.

Thanks for the input 👍